### PR TITLE
[MoE Refactor] Add sequence parallel tests to test_moe_layer.py

### PR DESCRIPTION
**Pull request by bnellnm**
## Purpose

Add sequence parallel tests to test_moe_layer.py.  Unfortunately, they require 4 gpus so I'm not sure we can run them in automation.  At least they can be run locally when validating changes to MoE.

cc @yzong-rh 

## Test Plan

Ran by hand with 4 gpus.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: tests/kernels/moe/test_moe_layer.py [hunk 1]
```diff
@@ -1448,7 +1553,7 @@ def _run_one_config(
         else:
             atol, rtol = 3.5e-2, 3.5e-2
     elif quantization in ("fp8", "fp8_blocked", "modelopt_fp8"):
-        atol, rtol = 6e-2, 6e-2
+        atol, rtol = 6.5e-2, 6.5e-2
```
Comment (yzong-rh) [inline tests/kernels/moe/test_moe_layer.py:1552]: Note: tolerance changed.

File: tests/kernels/moe/test_moe_layer.py [hunk 2]
```diff
@@ -1015,6 +1083,7 @@ def make_fake_moe_layer(
     tp_size: int = 1,
     dp_size: int = 1,
     ep_size: int = 1,
+    is_sequence_parallel: bool = False,
```
Comment (yzong-rh) [inline tests/kernels/moe/test_moe_layer.py:1084]: Note: `is_sequence_parallel` is not used in `make_fake_moe_layer`

File: tests/kernels/moe/test_moe_layer.py [hunk 3]
```diff
@@ -272,6 +312,15 @@ class MoETestConfig:
...
 
+    @property
+    def is_sequence_parallel(self) -> bool:
+        # Sequence parallelism: EP enabled + TP dimension used for sequence splitting
+        # In test config: ep_size represents total expert parallel size
+        # tp_size represents the original TP dimension (becomes sp_size in FusedMoE)
+        # dp_size represents data parallel size
+        # For SP: we need EP enabled (ep_size > 1) and sequence splitting (tp_size > 1)
+        return self.ep_size > 1 and self.tp_size > 1
+
```
Comment (yzong-rh) [inline tests/kernels/moe/test_moe_layer.py:327]: This works for
```
[2, 2, True],  # sequence parallel
```
added to `PARALLEL_COMBOS` but it's a bit confusing.

SP doesn't always need to be on when `EP > 1 and TP > 1`. Adding `use_sp` to `PARALLEL_COMBOS` would be more explicit.

File: tests/kernels/moe/test_moe_layer.py [hunk 4]
```diff
@@ -163,6 +166,43 @@ def override_normalize_e4m3fn_to_e4m3fnuz():
...
+            router_logits: torch.Tensor,
+        ) -> torch.Tensor:
+            # Split sequence across TP ranks
+            # Both hidden_states and router_logits have [num_tokens, ...] shape
+            hidden_states = sequence_parallel_chunk(hidden_states)
+            router_logits = sequence_parallel_chunk(router_logits)
+            # Run MoE on local chunk
+            result = fn(hidden_states, router_logits)
+            # Gather results from all TP ranks
+            return tensor_model_parallel_all_gather(result, 0)
```
Comment (yzong-rh) [inline tests/kernels/moe/test_moe_layer.py]: AI suggests adding a trim op after:
```
gathered = tensor_model_parallel_all_gather(result, 0)
return gathered[:hidden_states.shape[0]]
```
because `sequence_parallel_chunk` can add padding.

Although, if the untrimmed tensors are close the trimmed tensors should be as well.

**Review by yzong-rh — APPROVED** [review]
Also passed for me on 4xB200 on `0.20.2rc1.dev56+gf65376125.precompiled` after merging with commit `f6537612521df1156d8ac13f524e427aca908322` on main.

File: tests/kernels/moe/test_moe_layer.py (hunk 2 above)
Comment (bnellnm) [reply tests/kernels/moe/test_moe_layer.py:1084]: That's intentional.  The "fake" implementation is naive and doesn't really do any parallelization.

File: tests/kernels/moe/test_moe_layer.py (hunk 3 above)
Comment (bnellnm) [reply tests/kernels/moe/test_moe_layer.py:327]: I'm going by the definition of `ParallelConfig.use_sequence_parallel_moe`.  I'll add a comment.

File: vllm/distributed/device_communicators/all2all.py [hunk 5]
```diff
@@ -335,6 +335,7 @@ class NixlEPAll2AllManager(All2AllManagerBase):
     _lock = threading.Lock()
 
     def __init__(self, cpu_group, tcp_store_group=None):
+        assert tcp_store_group is not None
```
Comment (yzong-rh) [inline vllm/distributed/device_communicators/all2all.py:338]: Was this a pre-commit fix? It seems `tcp_store_group` can legitimately be `None` and is handled by the `All2AllManagerBase` superclass.

File: vllm/model_executor/layers/fused_moe/prepare_finalize/flashinfer_nvlink_two_sided.py [hunk 6]
```diff
@@ -185,13 +185,14 @@ def flashinfer_alltoall_dispatch(
...
         )
 
-        x_sf = MnnvlMoe.mnnvl_moe_alltoallv(
-            x_sf,
-            alltoall_info,
-            all2all_manager.workspace_tensor,  # type: ignore[attr-defined]
-            ep_rank,
-            ep_size,
-        )
+        if x_sf is not None:
```
Comment (yzong-rh) [inline vllm/model_executor/layers/fused_moe/prepare_finalize/flashinfer_nvlink_two_sided.py:188]: Weird, `pre-commit` and `pre-commit run --hook-stage manual mypy-3.13 --all-files` all succeed for me...
Couldn't reproduce https://github.com/vllm-project/vllm/actions/runs/25584737631/job/75110965865

File: vllm/distributed/device_communicators/all2all.py (hunk 5 above)
Comment (bnellnm) [reply vllm/distributed/device_communicators/all2all.py:338]: Nixl needs `tcp_store_group` to be non-None.  It crashes at runtime if not set here.  This assert will catch it earlier.

File: vllm/model_executor/layers/fused_moe/prepare_finalize/flashinfer_nvlink_two_sided.py (hunk 6 above)
Comment (bnellnm) [reply vllm/model_executor/layers/fused_moe/prepare_finalize/flashinfer_nvlink_two_sided.py:188]: I fixed this with the `fix lint` commit.
